### ISSUE TITLE
Deprecate MachineHealthCheck MaxUnhealthy and UnhealthyRange

### ISSUE THREAD

**Issue by vincepri** [kubernetes-sigs/cluster-api#10722]
MachineHealthCheck currently exposes these fields:

```go
        // Any further remediation is only allowed if at most "MaxUnhealthy" machines selected by
	// "selector" are not healthy.
	// +optional
	MaxUnhealthy *intstr.IntOrString `json:"maxUnhealthy,omitempty"`

	// Any further remediation is only allowed if the number of machines selected by "selector" as not healthy
	// is within the range of "UnhealthyRange". Takes precedence over MaxUnhealthy.
	// Eg. "[3-5]" - This means that remediation will be allowed only when:
	// (a) there are at least 3 unhealthy machines (and)
	// (b) there are at most 5 unhealthy machines
	// +optional
	// +kubebuilder:validation:Pattern=^\[[0-9]+-[0-9]+\]$
	UnhealthyRange *string `json:"unhealthyRange,omitempty"`
```

At a first glance, the fields seems to control remediation, and the comment seems to suggest as such; although in reality they only control setting the _conditions_ not when health checks fail and remediation should occur. This can be a confusing behavior to most users and counter intuitive at best.

For example:

Let's say I have 10 machines in my cluster, and 5 go unhealthy for some reason. The knobs above, if set let's say to only allow 20% or 2, make the MachineHealthCheck to stop setting the condition after 2 machines have been marked and continue if and only if the rest of the Machines. In reality, 5 Machines are unhealthy, but only 2 are marked as such.

**Comment by enxebre** [comment]
Related https://github.com/kubernetes-sigs/cluster-api/issues/5291

**Comment by sbueringer** [comment]
+1 having configuration that restricts how many machines can be marked as unhealthy seems strange. Instead we should have config restricting remediation (like the recently introduced remediation strategy on MD)

**Comment by enxebre** [comment]
>+1 having configuration that restricts how many machines can be marked as unhealthy seems strange. Instead we should have config restricting remediation (like the recently introduced remediation strategy on MD)

Agreed. The current API, while serving well, is more reminiscent of the original MHC design, where default remediation was built into the MHC itself.
Although to make sense of upcoming changes/deprecations I think we'll need to collect community feedback to check if there's any valid scenario where scalable resources might be decoupled from the short-circuiting remediation group as enabled by the current API. E.g. there are MSs per az but one single maxUnavailable budget.

**Comment by fabriziopandini** [comment]
Reporting an interesting [comment](https://github.com/kubernetes-sigs/cluster-api/pull/10853#issuecomment-2228264050) from #10853 

> My only concern before actually drop them in future is to make sure don't break any consumer where scalable resources might be decoupled from the short-circuiting remediation group as enabled by these fields today. E.g. there are MSs per az but one single common maxUnavailable budget for them.

That's an interesting note.
Might be what we need is to be more explicit about the fact that those fields are about short-circuiting remediation (e.g by moving them in a sub struct + amending documentation)


Considering that in both cases (removal or renaming) we need to go through deprecation of current fields

**Comment by sbueringer** [comment]
We had some discussions over the last few months and the outcome is that we want to keep the fields (so we are going to remove the deprecations) but we are going to rename them in v1beta2 so they are more intuitive.

See https://github.com/kubernetes-sigs/cluster-api/issues/12497 for more details. PR will be up in the next few days.

So I'm going to close this issue as we are not going to remove the fields anymore.

/close

(cc @fabriziopandini @JoelSpeed)
